### DISCUSSION TITLE
WIX6818 When using DotNetCoreBootstrapperApplicationHost, the Payload element for the BA's entry point DLL must have bal:BAFactoryAssembly="yes".

### DISCUSSION THREAD

**Discussion by TheOpenDevProject** [wixtoolset/issues#7276] [Questions] [answered]
Hey guys,

I am very new to Wix4.

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal">
  <Bundle Name="ForceConsoleInstallerBundle" Manufacturer="TODO Manufacturer" Version="1.0.0.0" UpgradeCode="5ffefd29-d990-42d8-8cff-2ec3b635186f">
	  <bal:Condition
			  Message="This application requires .NET 6 runtime. Please install the runtime then run this installer again."
			  Condition="[NetCheck=0]" />
	  <BootstrapperApplication>
		  <bal:WixDotNetCoreBootstrapperApplicationHost AlwaysInstallPrereqs="no"/>
    </BootstrapperApplication>

    <Chain>
		<PackageGroupRef Id="NetCoreInstaller"/>
      <!-- TODO: Define the list of chained packages. -->
       <MsiPackage SourceFile="ForceConsoleInstaller.msi" bal:PrereqPackage="yes" Id="UniversalInstaller" Vital="yes"/>
	
    </Chain>

  </Bundle>
</Wix>
```

I am trying to get .NET core installed before the Msi package is installed (Or after it actually doesn't matter).

Could someone help me understand what the error message I am getting is like I am 5 years old.

Cheers.

**Comment by barnson** [comment, marked as answer]
`WixDotNetCoreBootstrapperApplicationHost` is for when you're providing a custom BA. If you aren't, you might try https://wixtoolset.org/docs/reference/schema/bal/wixstandardbootstrapperapplication/ instead.

**Reply by TheOpenDevProject** [reply]
I'm going to mark this as the partial answer and leave a working solution below now I have figured it out.

**Comment by TheOpenDevProject** [comment]
I think its fairly urgent to start writing tutorials - the documentation isn't clear in most instances as to how to achieve common tasks like this.

Once I figure out how to do what I want, I would be happy to contribute.

**Reply by robmen** [reply]
There are many examples in the [Burn integration testing.](https://github.com/wixtoolset/wix4/tree/develop/src/test/burn). That would be a great place to start learning so you can start the tutorial.

**Reply by TheOpenDevProject** [reply]
I have figured it out after about 2 and a half days of probing and googling.

How would you like me to contribute a step-by-step, also I can't believe how responsive you are :heart:

**Reply by robmen** [reply]
Start by finding a logical place in the [current documentation](https://wixtoolset.org/docs/) to add content to or pages before/after. Then ask if that's a good place on the [WiX Development](https://github.com/orgs/wixtoolset/discussions/categories/wix-development). After that, it's a matter of writing the pieces of information that you found were necessary for someone to be successful.

Everything here is easy for me because I've written or reviewed the code itself. You have much more insight as to what a beginner needs to know. That's why I have David with me in the [Deployment Dojo](https://youtube.com/playlist?list=PLDlzbQXIs18slmqmdlS10_de_Cps-QRg6). :smile:

**Comment by TheOpenDevProject** [comment]
**For those who come across this in the near future**

1. Create a new Wix v4 Bundle if you don't have one already
2. Add a new Wix Source File into your bundle project and name it something like `NetCoreInstaller.wxs`
3. Add the following code to that file
```xml
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs" xmlns:netfx="http://wixtoolset.org/schemas/v4/wxs/netfx" xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal">
	<Fragment>
		<PackageGroup Id="NetCoreInstaller">
			<!-- Install .NET 6.0 -->
			<ExePackage Id="Netfx60"
						DisplayName="Microsoft .NET 7.0 SDK And Runtime"
						PerMachine="yes" Permanent="yes" Vital="yes" InstallArguments="/norestart /quiet"
						DetectCondition="DOT_NET_VER &gt;= v6.0.14"
						bal:PrereqPackage="yes">
				<ExePackagePayload
					Name="dotnet-sdk-7.0.202-win-x64.exe"
					DownloadUrl="https://download.visualstudio.microsoft.com/download/pr/c6ad374b-9b66-49ed-a140-588348d0c29a/78084d635f2a4011ccd65dc7fd9e83ce/dotnet-sdk-7.0.202-win-x64.exe"
					Hash="58e9437f2ead1e9edd67d837f7d2e1313d866d510629465b0485965a9540c65eddd6fffafad9cf3261cb476a1dfff20404908f3bf06a7e01bc3bebc0bcb27d39"
					Size="212900"
          />
				<ExitCode Value="0" Behavior="success"/>
				<ExitCode Behavior="scheduleReboot"/>
			</ExePackage>
		</PackageGroup>
	</Fragment>
</Wix>
```
A quick explanation of the XML node properties
Make sure you calculate a **SHA512** hash of the Exe (I did this by downloading it), and set the filesize the same as the installer file size in Kb

4. Finally add this into your Bundle.wxs file `<PackageGroupRef Id="NetCoreInstaller"/>` under the `<Chain>` section

**Reply by rseanhall** [reply]
I don't remember the name of the command off the top of my head, but it would be easier to use `wix.exe burn <commandname>` to get the hash information.
